embassy-stm32/src/dma/dma.rs

```diff
@@ -56,6 +57,93 @@ impl State {
 
 static STATE: State = State::new();
 
+struct M0AR(u32);
+struct M1AR(u32);
+struct ChunkSize(u16);
+
+struct GiantTransferState {
+    enabled: [AtomicBool; DMA_CHANNEL_COUNT],
```

Comment:
this would be more readable with a `ChannelState` struct with all the stuff for one channel, and then a have a `[ChannelState; DMA_CHANNEL_COUNT]`. Also fold it with the old STATE.